lib/screens/v2/widget_instance/train_crowding.ex

```diff
@@ -0,0 +1,90 @@
+defmodule Screens.V2.WidgetInstance.TrainCrowding do
+  @moduledoc """
+  A widget that displays the crowding on a train that is en route to the current station.
+  """
+
+  alias Screens.Config.Screen
+  alias Screens.Config.V2.Triptych
+  alias Screens.Predictions.Prediction
+
+  defstruct screen: nil,
+            prediction: nil,
+            now: nil
+
+  @type t :: %__MODULE__{
+          screen: Screen.t(),
+          prediction: Prediction.t(),
+          now: DateTime.t()
+        }
+
+  @type widget_data :: %{
+          destination: String.t(),
+          crowding: list(crowding_level),
+          platform_position: number,
+          front_car_direction: :left | :right,
+          now: String.t()
+        }
+
+  @type crowding_level :: :no_data | :not_crowded | :some_crowding | :crowded | :disabled
+
+  @spec serialize(t()) :: widget_data()
+  def serialize(%__MODULE__{
+        screen: %Screen{app_params: %Triptych{train_crowding: train_crowding}},
+        prediction: prediction,
+        now: now
+      }) do
+    %{
+      destination: prediction.trip.headsign,
+      crowding: serialize_carriages(prediction.vehicle.carriages),
+      platform_position: train_crowding.platform_position,
+      front_car_direction: train_crowding.front_car_direction,
+      now: serialize_time(now)
+    }
+  end
+
+  defp serialize_time(%DateTime{} = time) do
+    DateTime.to_iso8601(time)
+  end
+
+  defp serialize_carriages(nil), do: nil
+
+  defp serialize_carriages(carriages),
+    do: Enum.map(carriages, fn car -> serialize_occupancy_status(car) end)
+
+  defp serialize_occupancy_status(:no_data_available), do: :no_data
+  defp serialize_occupancy_status(:many_seats_available), do: :not_crowded
+  defp serialize_occupancy_status(:few_seats_available), do: :not_crowded
+  defp serialize_occupancy_status(:standing_room_only), do: :some_crowding
+  defp serialize_occupancy_status(:crushed_standing_room_only), do: :crowded
+  defp serialize_occupancy_status(:full), do: :crowded
+  defp serialize_occupancy_status(:not_accepting_passengers), do: :disabled
```

Comment:
Nitpicky, but can this conversion live in the vehicle parser? I would imagine if we were to show crowding info for other lines, it would use the same conversion system. Total guess though so up to you.

Reply:
Hmm. In my comments to your previous comments, I mentioned my rationale here is that parsing should be a direct translation, while serialize handles "present it this way in this particular context". I can see us using this conversion in other situations, but I can also see us not doing that, and choosing to use all 5 crowding types. Maybe I'll ask design